### ISSUE TITLE
payinvoice bug

### ISSUE THREAD

**Issue by wxf4150** [omnilaboratory/lnd#4]
```
lncli   payinvoice obort2147483653:11p3tr9xnpp5f8zhjyktawpvcapn5c4pp27xrhzaua0mrm6emz9gcqs09y64r8ysdqqcqzpgxqyz5vq3q8zqqqqq9sp555j0jmrpnmztvu7gey0sw7lph3r5s00xsynu7n6z7k7zv54a0yyq9qyyssqmqwt3l9axt3hjefas8jnqn5f37v5ggneds76wvuqh8mq0yu4fy4ndyelaud7r27st3009fyn4pry27rjs5u3ws82neuengxgwsheuyqqtcmvh9
Payment hash: 49c57912cbeb82cc7433a62a10abc61dc5de75fb1ef59d88a8c020f2935519c9
Description: 
Amount (in satoshis): 100000000
Fee limit (in satoshis): 5000000
Destination: 03202f05bde97faf6675dda34f9e43932e3a56d7a54c12f8dd02b8ffcf40bddc7a
Confirm payment (yes/no): yes
+------------+--------------+--------------+--------------+-----+----------+----------+-------+
| HTLC_STATE | ATTEMPT_TIME | RESOLVE_TIME | RECEIVER_AMT | FEE | TIMELOCK | CHAN_OUT | ROUTE |
+------------+--------------+--------------+--------------+-----+----------+----------+-------+
+------------+--------------+--------------+--------------+-----+----------+----------+-------+
Amount + fee:   0 + 0 sat
Payment hash:   49c57912cbeb82cc7433a62a10abc61dc5de75fb1ef59d88a8c020f2935519c9
Payment status: FAILED, reason: FAILURE_REASON_NO_ROUTE
[lncli] FAILED
```

**Comment by wxf4150** [comment]
> Payment status: FAILED, reason: FAILURE_REASON_NO_ROUTE
the err fix with #5 

new err:

2022-06-22 18:03:58.988 [ERR] HSWC: ChannelLink(86d38ad45882528d6a440f2475df3d996e6ee8a3f6175d8fe75bda9bea20f3de:1): failing link: unable to update commitment: height=1, for ChannelPoint(86d38ad45882528d6a440f2475df3d996e6ee8a3f6175d8fe75bda9bea20f3de:1) attempts to consume 0.0002066 BTC while channel capacity is 0.0002 BTC with error: internal error.

the fee is 20660 sat, may the fee is err. but i will funding more btc for a test.

**Comment by wxf4150** [comment]
0.0002066  -0.0002 =660 .
it 's tow Anchor output.   the anchor have 330 sat.
i disable the anchor output .  the erro gone.   commit at 599bee90d79a740c39c5cdf859fe962232372d94

**Comment by wxf4150** [comment]
new err .  

```
lncli   payinvoice obort2147483653:11p3t970rpp5jxfwpepa2qd93552r2kp2zjqj670pdlrl4fqcsscyn9ryta28sssdqqcqzpgxqyz5vq3q8zqqqqq9sp5s7ccq75dee3a3nh48dhd68dagf4v9adyfzz4j8c28r8kpu6y590s9qyyssqhuuemktka9xy997qm052xg8gdj6uz2qy972f2x0adeuncn9c7x0xx9dwc33x5m0433w903jh59ml0edlhjkauhmy9hs50vs2qqm6fscqwqdcyl
Payment hash: 9192e0e43d501a58d28a1aac150a4096bcf0b7e3fd520c421824ca322faa3c21
Description: 
AssetId : 2147483653
Amount (in satoshis): 100000000
Fee limit (in satoshis): 5000000
Destination: 03202f05bde97faf6675dda34f9e43932e3a56d7a54c12f8dd02b8ffcf40bddc7a
Confirm payment (yes/no): yes
+------------+--------------+--------------+--------------+-----+----------+-----------------+----------------------+
| HTLC_STATE | ATTEMPT_TIME | RESOLVE_TIME | RECEIVER_AMT | FEE | TIMELOCK | CHAN_OUT        | ROUTE                |
+------------+--------------+--------------+--------------+-----+----------+-----------------+----------------------+
| IN_FLIGHT  |        4.294 |            - | 100000       | 0   |      293 | 266081813987329 | 03202f05bde97faf6675 |
+------------+--------------+--------------+--------------+-----+----------+-----------------+----------------------+
Amount + fee:   0 + 0 sat
Payment hash:   9192e0e43d501a58d28a1aac150a4096bcf0b7e3fd520c421824ca322faa3c21
Payment status: IN_FLIGHT

```
it's blocked .

**Comment by wxf4150** [comment]
i had open  autopilot
the new err
```
2022-06-23 17:26:38.252 [DBG] PEER: Received UpdateAddHTLC(chan_id=f087948b75eb120dbef69f02a8deca63a4d66d9bb3a23b281e58e7866c282cea, id=0, amt=100000000, expiry=308, hash=bcfa35d9c9839ba10175b5b5bf4785348a17b336bb3cdfddd8f392e7dd6788ee) from 0328031e59d805f01459699665cf7bfc89af624eeee9f16ded9d78aebf5caeec3c@127.0.0.1:58412
2022-06-23 17:26:38.307 [DBG] PEER: Received CommitSig(chan_id=f087948b75eb120dbef69f02a8deca63a4d66d9bb3a23b281e58e7866c282cea, num_htlcs=1) from 0328031e59d805f01459699665cf7bfc89af624eeee9f16ded9d78aebf5caeec3c@127.0.0.1:58412
2022-06-23 17:26:38.308 [ERR] HSWC: ChannelLink(eb2c286c86e7581e283ba2b39b6dd6a463cadea8029ff6be0d12eb758b9487f0:1): failing link: ChannelPoint(eb2c286c86e7581e283ba2b39b6dd6a463cadea8029ff6be0d12eb758b9487f0:1): unable to accept new commitment: rejected commitment: commit_height=1, invalid_commit_sig=304402203d8fd9644d293fddbf30097d03180a4002428393b285cef62f2921e6b444c753022001e6c55721c15c7930bd488e26117854eccced0e57c642934b9ef64ea6028159, commit_tx=0200000001f087948b75eb120dbef69f02a8deca63a4d66d9bb3a23b281e58e7866c282ceb01000000003b0b2780030000000000000000216a1f6f6d6e69000000078000000502020000000017d78400010000000005f5e100660600000000000017a91456e1c726ffc5bbea4c141c37ea9b111ef64496db8772711e000000000017a914f0729f19b882d449c882ffe989bb15e7bc75ce5b87fd4beb20, sig_hash=6af0756bda2161637aaaac36cedad31cd850576acef7ae8293dc0b91564bdf20 with error: invalid commitment
2022-06-23 17:26:38.308 [ERR] HSWC: ChannelLink(eb2c286c86e7581e283ba2b39b6dd6a463cadea8029ff6be0d12eb758b9487f0:1): link failed, exiting htlcManager
```

after UpdateAddHTLC event,  when verify commitment.sig  got some err.

**Comment by wxf4150** [comment]
>rejected commitment: commit_height=1, invalid_commit_sig=304402
have fix at 9aa90f6b5d00914effd9aefefdf4a3e4b7076e04

new err: tlv stream is not canonical with error: internal error

server log:
```
2022-06-24 16:07:22.194 [DBG] PEER: Received UpdateAddHTLC(chan_id=96bc19e2a11adf0bd3df8db80553ef00aa2bbae023bfa5965f41db47a022a062, id=0, amt=100000000, expiry=356, hash=1aa5660981a2521bd4803ed3716d138c553bac72f8bec9532b73b57d43ea4395) from 0328031e59d805f01459699665cf7bfc89af624eeee9f16ded9d78aebf5caeec3c@127.0.0.1:58580
2022-06-24 16:07:22.248 [DBG] PEER: Received CommitSig(chan_id=96bc19e2a11adf0bd3df8db80553ef00aa2bbae023bfa5965f41db47a022a062, num_htlcs=1) from 0328031e59d805f01459699665cf7bfc89af624eeee9f16ded9d78aebf5caeec3c@127.0.0.1:58580
2022-06-24 16:07:22.251 [DBG] PEER: Sending RevokeAndAck(chan_id=96bc19e2a11adf0bd3df8db80553ef00aa2bbae023bfa5965f41db47a022a062, rev=b5d8c9a2b12ee5ebef308d82ab5639b890fee9f9fdc00b418f9f623422bf2a07, next_point=0398c7ce80b1520c7f99d52dbae112ca29c78134040e904e439b6a672bca7bd55c) to 0328031e59d805f01459699665cf7bfc89af624eeee9f16ded9d78aebf5caeec3c@127.0.0.1:58580
2022-06-24 16:07:22.255 [DBG] PEER: Sending CommitSig(chan_id=96bc19e2a11adf0bd3df8db80553ef00aa2bbae023bfa5965f41db47a022a062, num_htlcs=1) to 0328031e59d805f01459699665cf7bfc89af624eeee9f16ded9d78aebf5caeec3c@127.0.0.1:58580
2022-06-24 16:07:22.267 [DBG] PEER: Received RevokeAndAck(chan_id=96bc19e2a11adf0bd3df8db80553ef00aa2bbae023bfa5965f41db47a022a062, rev=264a06439b340d19aa2338ff950cab9026e48658d0527d732a7730772dbd8ffb, next_point=026d93fa72871e9c7a1e732b1acc2a77e697a944f6200e8b553c5c79af81f63e14) from 0328031e59d805f01459699665cf7bfc89af624eeee9f16ded9d78aebf5caeec3c@127.0.0.1:58580
2022-06-24 16:07:22.282 [DBG] INVC: Invoice(pay_hash=1aa5660981a2521bd4803ed3716d138c553bac72f8bec9532b73b57d43ea4395, pay_addr=884a62e613fb2146d1f092c8a1947a11da692b057913e71e5dc9e9e6e93029f2): tlv stream is not canonical, amt=100000000, expiry=356, circuit=(Chan ID=308:1:1, HTLC ID=0), mpp=total=100000000, addr=884a62e613fb2146d1f092c8a1947a11da692b057913e71e5dc9e9e6e93029f2, amp=<nil>
2022-06-24 16:07:22.282 [ERR] HSWC: ChannelLink(63a022a047db415f96a5bf23e0ba2baa00ef5305b88ddfd30bdf1aa1e219bc96:1): failing link: tlv stream is not canonical with error: internal error
2022-06-24 16:07:22.282 [ERR] HSWC: ChannelLink(63a022a047db415f96a5bf23e0ba2baa00ef5305b88ddfd30bdf1aa1e219bc96:1): link failed, exiting htlcManager
2022-06-24 16:07:22.282 [INF] HSWC: ChannelLink(63a022a047db415f96a5bf23e0ba2baa00ef5305b88ddfd30bdf1aa1e219bc96:1): exited
```

**Comment by wxf4150** [comment]
> tlv stream is not canonical with error: internal error
fix at commit  7f0214bf60be31b31b8ec1860f50cce9f2b8c888

 the payinvoice cmd ok now
```
lncli  payinvoice obort2147483653:11p3t2aampp5ksudcxa7tt46ewrv6t9a5ppxhpmad0595tn4gufwrfqe7rddjmkqdqqcqzpgxqyz5vq3q8zqqqqq9sp5rhh8dqr85sqd9j0fdy7fgtzfxs25xf6w3jfm0h8etty68fydteps9qyyssqxtptfqgmctqsh2akveqjhx2rfcx3w37x5muqnd60pw5kqpeqv7dru2a2necrde96y4wqpdhuelqpqtu268kp4dtx62mujq5ph60zxpqp7hrklu
Payment hash: b438dc1bbe5aebacb86cd2cbda0426b877d6be85a2e754712e1a419f0dad96ec
Description: 
AssetId : 2147483653
Amount (in satoshis): 100000000
Fee limit (in satoshis): 5000000
Destination: 03202f05bde97faf6675dda34f9e43932e3a56d7a54c12f8dd02b8ffcf40bddc7a
Confirm payment (yes/no): yes
+------------+--------------+--------------+--------------+-----+----------+-----------------+-------+
| HTLC_STATE | ATTEMPT_TIME | RESOLVE_TIME | RECEIVER_AMT | FEE | TIMELOCK | CHAN_OUT        | ROUTE |
+------------+--------------+--------------+--------------+-----+----------+-----------------+-------+
| SUCCEEDED  |        0.032 |        0.144 | 100000       | 0   |      362 | 345246651187201 |       |
+------------+--------------+--------------+--------------+-----+----------+-----------------+-------+
Amount + fee:   100000 + 0 sat
Payment hash:   b438dc1bbe5aebacb86cd2cbda0426b877d6be85a2e754712e1a419f0dad96ec
Payment status: SUCCEEDED, preimage: 974d85bc1a569ef17a197a033dfcacd0c58a9d957278b7c7aa3944c394cd24c3
```
